app/flatpak-builtins-install.c

```diff
@@ -299,10 +300,15 @@ flatpak_builtin_install (int argc, char **argv, GCancellable *cancellable, GErro
   if (!flatpak_split_partial_ref_arg (name, &opt_arch, &branch, error))
     return FALSE;
 
+  if (branch != NULL && *branch != 0)
+    actual_branch = g_strdup (branch);
+  else
+    actual_branch = flatpak_dir_get_remote_default_branch (dir, repository);
+
   if (!opt_app && !opt_runtime)
     opt_app = opt_runtime = TRUE;
 
-  ref = flatpak_dir_find_remote_ref (dir, repository, name, branch, opt_arch,
+  ref = flatpak_dir_find_remote_ref (dir, repository, name, actual_branch, opt_arch,
                                      opt_app, opt_runtime, &is_app, cancellable, error);
```

Comment:
I've updated this particular commit following this suggestion, tested it and it seems to be working now as expected (again, good catch. The previous one did fail in the case you mentioned).

See the new commit here (already force-pushed into this PR):
https://github.com/flatpak/flatpak/pull/345/commits/2541910dcb6d2d7e0f9347676df33117d0bf702b